### ISSUE TITLE
dynamic? doesn't really properly check whether a var is dynamic

### ISSUE THREAD

**Issue by davisn** [positronic-solutions/pulley.cps#14]
`dynamic?` is currently written as:

```
(defn dynamic? [resolved-var]
  (:dynamic (meta resolved-var)))
```

However, not all dynamic vars have `:dynamic` set in their meta-data.  Attempts to dynamically bind these vars, e.g. `*out*`, results in an error:

```
(cps/cps (with-out-str (println "foo")))

clojure.lang.Compiler$CompilerException: java.lang.IllegalStateException: Can't dynamically bind non-dynamic var: #'clojure.core/*out*, compiling:(NO_SOURclojure.lang.Compiler$CompilerException: java.lang.IllegalStateException: Can't dynamically bind non-dynamic var: #'clojure.core/*out*, compiling:(NO_SOUR
```

**Comment by davisn** [comment]
It seems the proper way to perform this check would be with the `isDynamic()` method on `Var`.

**Comment by davisn** [comment]
Fixed in d3dbbe104ba4a934ad9899214beeed0ac16dc26d
